### DISCUSSION TITLE
Inline Javascript gets encoded when using editor.getSelected().toHTML()

### DISCUSSION THREAD

**Discussion by stljeff1** [GrapesJS/grapesjs#3776] [Q&A] [answered]
I am hoping to gain some insight and perspective on using `editor.getSelected().toHTML()`.

I am using Monaco for my main code editor in my app, but I did not change the popup editor. Now that I like Monaco editor, I want to use it for the popup editor that pops up when editing the HTML for a component.

Anyway, I have a custom widget that is the popup containing my code editor. I also have an edit button on the component's toolbar that will open the popup code editor. When that button is pressed, I grab the HTML of the component's HTML and pass it to my popup widget. I grab the HTML by calling `editor.getSelected().toHTML()`

As far as I can tell, calling this function will character encode my inline javascript. At first I thought maybe Monaco Editor was encoding this, but logging the returned value from editor.getSelected().toHTML() confirms the javascript is encoded before triggering my popup.

Also, I recently upgraded grapesjs from 0.16.12 to 0.16.34. Did the implementation of toHTML change in those versions? Does GrapesJS normally encode javascript, and perhaps I undid some hack that was in place? If toHTML() is not encoding my javascript, what could be the culprit? I instantiate Grapes with non-encoded javascript.

**Comment by artf** [comment, marked as answer]
The text encoding should been fixed here https://github.com/artf/grapesjs/releases/tag/v0.17.22

**Reply by stljeff1** [reply]
thanks for confirming.
